pageserver/src/tenant/gc_block.rs

```diff
@@ -42,6 +43,24 @@ impl GcBlock {
         }
     }
 
+    /// Blocks GC until `duration` has elapsed.
+    ///
+    /// We do this as the leases mapping are not persisted to disk. By delaying GC by default
+    /// length, we guarantee that all the leases we granted before will expire when we run GC for
+    /// the first time after restart / transition from AttachedMulti to AttachedSingle.
+    pub(super) async fn block_for(
+        &self,
+        duration: Duration,
+        cancel: &CancellationToken,
+    ) -> Result<(), Cancelled> {
```

Comment:
Feels like returning the `Cancelled` here is a bit weird since it's from a sibling module.

Reply:
I guess we would never have this overall problem in the first place if the gc background task would just wait until it can do something, instead of waiting to see if it can do something.